src/utils.jl

```diff
@@ -424,3 +424,49 @@ function dof_to_node(dh::Ferrite.AbstractDofHandler, u::Array{T,1}; field::Int=1
     end
     return mapslices(process, data, dims=[2])
 end
+
+get_gradient_interpolation(::Ferrite.Lagrange{dim,shape,order}) where {dim,shape,order} = Ferrite.DiscontinuousLagrange{dim,shape,order-1}()
+get_gradient_interpolation_type(::Type{Ferrite.Lagrange{dim,shape,order}}) where {dim,shape,order} = Ferrite.DiscontinuousLagrange{dim,shape,order-1}
+# TODO remove if Knuth's PR on this gets merged (Ferrite PR 552)
+getgrid(dh::Ferrite.DofHandler) = dh.grid
+
+"""
+Compute the piecewise discontinuous gradient field for `field_name`. Returns the flux dof handler and the corresponding flux dof values.
+"""
+function compute_gradient_field(dh::Ferrite.DofHandler{dim}, u::AbstractVector, field_name::Symbol) where {dim}
+    # Get 
+    field_idx = Ferrite.find_field(dh, field_name)
+    ip = Ferrite.getfieldinterpolation(dh, field_idx)
+
+    # Dof handler for gradient field
+    dh_gradient = Ferrite.DofHandler(getgrid(dh))
+    ip_gradient = get_gradient_interpolation(ip)
+    push!(dh_gradient, :gradient, Ferrite.getfielddim(dh, field_name)*dim, ip_gradient) # field dim × spatial dim components
+    Ferrite.close!(dh_gradient)
+
+    # Buffer for the dofs
+    cell_dofs = zeros(Int, Ferrite.getnbasefunctions(ip))
+    cell_dofs_gradient = zeros(Int, Ferrite.getfielddim(dh_gradient, :gradient))
+
+    # Allocate storage for the fluxes to store
+    u_gradient = zeros(Ferrite.ndofs(dh_gradient))
+
+    for (cell_num, cell) in enumerate(Ferrite.CellIterator(dh))
+        Ferrite.celldofs!(cell_dofs, dh, cell_num)
+        uᵉ = u[cell_dofs[Ferrite.dof_range(dh, field_name)]]
+
+        Ferrite.celldofs!(cell_dofs_gradient, dh_gradient, cell_num)
+        uᵉ_gradient = u_gradient[cell_dofs_gradient]
+
+        ref_coords = Ferrite.reference_coordinates(ip_gradient)
+        for i ∈ 1:Ferrite.getnbasefunctions(ip_gradient)
```

Comment:
Within this loop we need another for loop for the dimension of the field, something like this:
```
    for (cell_num, cell) in enumerate(Ferrite.CellIterator(dh))
        Ferrite.celldofs!(cell_dofs, dh, cell_num)
        uᵉ = u[cell_dofs[Ferrite.dof_range(dh, field_name)]]

        Ferrite.celldofs!(cell_dofs_gradient, dh_gradient, cell_num)
        uᵉ_gradient .= u_gradient[cell_dofs_gradient]

        ref_coords = Ferrite.reference_coordinates(ip_gradient)
        for i ∈ 1:Ferrite.getnbasefunctions(ip_gradient)
            _derivative = Ferrite.derivative(ip, ref_coords[i])
            for j ∈ 1:Ferrite.getnbasefunctions(ip)
                for d in 1:Ferrite.getfielddim(dh,field_name)
                    uᵉ_gradient[d*i] += uᵉ[j*d] * _derivative[j][d]
                end
            end 
        end
        u_gradient[cell_dofs_gradient] .= uᵉ_gradient
    end
    return (dh_gradient, u_gradient)
```
Either I messed something up with the indices access or maybe its by construction missing some dofs (probably the prior) ? The incompressible elasticity example with quadratic displacement looks like this:
![image](https://user-images.githubusercontent.com/27497290/207565775-3e5305da-200a-4467-9ba4-085a89a0549a.png)

The `getnbasefunctions` does not know about the dimensionality of the field and returns e.g. 

```julia
julia> Ferrite.getnbasefunctions(DiscontinuousLagrange{2,RefTetrahedron,1}())
3
```
which is true for scalar fields, for vector fields the dimensionality need to be taken into account e.g. `4` entries for 2D field in 2D space, which ends up in 12 dofs per cell

```julia
julia> Ferrite.getfielddim(dh_grad,:gradient)
4

julia> Ferrite.ndofs_per_cell(dh_grad)
12

```

Reply:
Resolved with latest commit and generator fix in https://github.com/Ferrite-FEM/Ferrite.jl/pull/557